### PR TITLE
Unittesting NodeInfo and NodeTreeManager

### PR DESCRIPTION
**Pull request by rayklauck**

### CODE REVIEW

File: src/labone/nodetree/node.py [hunk 1]
```diff
@@ -141,21 +141,6 @@ def path(self) -> str:
...
-    @property
-    def type(self) -> str:  # noqa: A003
-        """Type of the node."""
-        return self._info["Type"]
-
-    @property
-    def unit(self) -> str:
-        """Unit of the node."""
-        return self._info["Unit"]
-
```
Comment (tobiasah) [inline src/labone/nodetree/node.py:158]: issue: I know why you can remove this but its not user friendly. A user of the API does most likely not know that these keys are present in the node info. Therefore I would still hard-code them

File: tests/nodetree/test_node_unit.py [hunk 2]
```diff
@@ -21,12 +22,264 @@
...
+    def test_contains(self):
+        with patch.object(
+            NodeInfo,
+            "__dir__",
+            autospec=True,
+            return_value=["a"],
+        ) as dir_mock:
+            info = NodeInfo({})
+            assert "a" in info
+            dir_mock.assert_called_once_with(info)
```
Comment (tobiasah) [inline tests/nodetree/test_node_unit.py:47]: issue: This test does not test our API but python itself. Better use something like 

```suggestion
            info = NodeInfo({"a" : "Test"})
            assert "a" in info
```

**Review by tobiasah — CHANGES_REQUESTED** [review]

File: src/labone/nodetree/node.py (hunk 1 above)
Comment (rayklauck) [reply src/labone/nodetree/node.py:158]: I could write all viable options in the documentation of the class or the getattr - would that be acceptable?

File: tests/nodetree/test_node_unit.py (hunk 2 above)
Comment (rayklauck) [reply tests/nodetree/test_node_unit.py:47]: actually, the `in` operator calls the `__contains__` and we want to test that it calls `__dir__` , which is not trivial. I think the proposed changes make the test less of a unittest

**Review by tobiasah — APPROVED** [review]

**Pull request merged**